lib/cylc/task_events_mgr.py

```diff
@@ -578,6 +578,9 @@ def _process_job_logs_retrieval(self, schd_ctx, ctx, id_keys):
 
     def _job_logs_retrieval_callback(self, proc_ctx, schd_ctx):
         """Call back when log job retrieval completes."""
+        if proc_ctx.ret_code != 0:
+            LOG.warning("Log retrieval command exited with %d! Output: %s",
+                        proc_ctx.ret_code, proc_ctx.err)
```

Comment:
The `proc_ctx` has a `__str__` method with all the bells and whistles, so you should be able to do:

```python
if proc_ctx.ret_code:
    LOG.error(proc_ctx)
```

And perhaps even add an else for debug mode?

```python
else:
    LOG.debug(proc_ctx)
```

There is an example here (for job submit/poll/kill commands): https://github.com/cylc/cylc/blob/682424c5978bf966b75d4b004a2f1c83ed571ea7/lib/cylc/task_job_mgr.py#L480-L483

Reply:
Agreed! Updating branch in a few minutes. Took a bit longer as I wanted to give it a try, and my brain is still warming-up after these long holidays.

Before:

```bash
2019-01-02T22:29:24Z WARNING - Log retrieval command exited with 1! Output: rsync: Invalid argument passed to --chmod (Du=rwx,Dgo=rx,)
	rsync error: syntax or usage error (code 1) at main.c(1572) [client=3.1.1]
```

With the code suggested:

```bash
2019-01-02T22:32:52Z ERROR - [TaskJobLogsRetrieveContext(key='job-logs-retrieve', ctx_type='job-logs-retrieve', user_at_host='pbs', max_size=None) cmd] rsync -v -rltgoD --chmod=Du=rwx,Dgo=rx, Fu=rw,Fgo=r '--rsh=ssh -oBatchMode=yes -oConnectTimeout=10' --include=/1 --include=/1/a --include=/1/a/01 '--include=/1/a/01/**' '--exclude=/**' 'pbs:$HOME/cylc-run/pbs1/log/job/' /home/testuser/cylc-run/pbs1/log/job/
	[TaskJobLogsRetrieveContext(key='job-logs-retrieve', ctx_type='job-logs-retrieve', user_at_host='pbs', max_size=None) ret_code] 1
	[TaskJobLogsRetrieveContext(key='job-logs-retrieve', ctx_type='job-logs-retrieve', user_at_host='pbs', max_size=None) err]
	rsync: Invalid argument passed to --chmod (Du=rwx,Dgo=rx,)
	rsync error: syntax or usage error (code 1) at main.c(1572) [client=3.1.1]
```

More information that might be helpful when troubleshooting similar issues.

Thanks for the sharp eyes!

Reply:
Oh, and of course running without the error in the log retrieval command, everything works fine. Adding `--debug` to the command shows the extra `LOG.debug` as:

```bash
2019-01-02T22:35:56Z DEBUG - Performing suite health check
2019-01-02T22:35:56Z DEBUG - [TaskJobLogsRetrieveContext(key='job-logs-retrieve', ctx_type='job-logs-retrieve', user_at_host='pbs', max_size=None) cmd] rsync -v -rltgoD --chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r '--rsh=ssh -oBatchMode=yes -oConnectTimeout=10' -v --include=/1 --include=/1/a --include=/1/a/01 '--include=/1/a/01/**' '--exclude=/**' 'pbs:$HOME/cylc-run/pbs1/log/job/' /home/testuser/cylc-run/pbs1/log/job/
	[TaskJobLogsRetrieveContext(key='job-logs-retrieve', ctx_type='job-logs-retrieve', user_at_host='pbs', max_size=None) ret_code] 0
	[TaskJobLogsRetrieveContext(key='job-logs-retrieve', ctx_type='job-logs-retrieve', user_at_host='pbs', max_size=None) out]
	opening connection using: ssh -oBatchMode=yes -oConnectTimeout=10 pbs rsync --server --sender -vvlogDtre.iLsfx . "$HOME/cylc-run/pbs1/log/job/"  (10 args)
	receiving incremental file list
	[sender] showing directory 1 because of pattern /1
	delta-transmission enabled
	[sender] showing directory 1/a because of pattern /1/a
	[sender] hiding directory 1/b because of pattern /**
	[sender] showing directory 1/a/01 because of pattern /1/a/01
	[sender] hiding file 1/a/NN because of pattern /**
	[sender] showing file 1/a/01/job.xtrace because of pattern /1/a/01/**
	[sender] showing file 1/a/01/job.err because of pattern /1/a/01/**
	[sender] showing file 1/a/01/job-activity.log because of pattern /1/a/01/**
	[sender] showing file 1/a/01/job because of pattern /1/a/01/**
	[sender] showing file 1/a/01/job.out because of pattern /1/a/01/**
	[sender] showing file 1/a/01/job.status because of pattern /1/a/01/**
	1/a/01/job is uptodate
	1/a/01/job-activity.log is uptodate
	1/a/01/job.err is uptodate
	1/a/01/job.out is uptodate
	1/a/01/job.status is uptodate
	1/a/01/job.xtrace is uptodate
	total: matches=0  hash_hits=0  false_alarms=0 data=0
	
	sent 100 bytes  received 1,015 bytes  2,230.00 bytes/sec
	total size is 12,681  speedup is 11.37
2019-01-02T22:35:56Z DEBUG - ['cylc', 'jobs-submit', '--debug', '--', '/home/testuser/cylc-run/pbs1/log/job', '1/b/01']
```